### PR TITLE
noVNC Customization

### PR DESCRIPTION
**Pull request by pawan-idrive**
This PR contains the changes for having noVNC customization.

### CODE REVIEW

File: vendor/idrive_vnc_base.css [hunk 1]
```diff
@@ -0,0 +1,85 @@
...
+  box-sizing: border-box;
+}
+
+#screen {
+  flex: 1;
+  /* fill remaining space */
+  overflow: hidden
+}
+
+#canvasCtnr {
```
Comment (diganta-dutta) [inline vendor/idrive_vnc_base.css]:
```suggestion
#screen div {
```

File: vendor/idrive_vnc_base.css [hunk 2]
```diff
@@ -0,0 +1,85 @@
...
+  display:flex;
+  width:100%;
+  height: calc(100vh - 56px);
+  overflow:auto;
+  background:#000 !important;
+}
+
+#canvasArea {
```
Comment (diganta-dutta) [inline vendor/idrive_vnc_base.css]:
```suggestion
#screen div canvas
```

File: vnc_auto.html [hunk 3]
```diff
@@ -81,11 +44,19 @@
...
         // When this function is called we have
         // successfully connected to a server
         function connectedToServer(e) {
             status("Connected to " + desktopName);
+            screen_div = rfb._screen;
+            screen_div.id = "canvasCtnr";
+            screen_canvas = rfb._screen.firstChild
+            screen_canvas.id = 'canvasArea';
+            screen_canvas.style.margin = null;
+            rfb.focus();
```
Comment (diganta-dutta) [inline vnc_auto.html:52]: Move into our own js `idrive_vnc_base.js` and include in body.

**Review by diganta-dutta — CHANGES_REQUESTED** [review]

**Comment by pawan-idrive** [comment]
@diganta-dutta I have made CSS related changes but javascript changes didn't work as expected, therefore I have not introduced a new file yet.

File: vnc_auto.html (hunk 3 above)
Comment (pawan-idrive) [reply vnc_auto.html:52]: @diganta-dutta CSS related changes have been removed but `rfb.focus();` function is getting called in `connect` event. I have tried putting the same in custom JS file but things didn't work as expected
- I tried to get the DOM elements but it seems RFB object construction and then connect function takes time and the DOM is not ready yet so I am not able to set the focus
- I tried other ways too but didn't help
Overall, I would say for now, lets keep this change here only

Comment (diganta-dutta) [reply vnc_auto.html:52]: Ok

File: vendor/idrive_vnc_base.css [hunk 4]
```diff
@@ -0,0 +1,86 @@
...
+	bottom: 0;
+	left: 0;
+}
+
+#status {
+  text-align: center;
+  font-size: 14px;
+}
+
+#sendCtrlAltDelButton {
```
Comment (diganta-dutta) [inline vendor/idrive_vnc_base.css:36]: Is this there?

**Review by diganta-dutta — APPROVED** [review]

File: vendor/idrive_vnc_base.css (hunk 4 above)
Comment (diganta-dutta) [reply vendor/idrive_vnc_base.css:36]: @pawan-idrive Please address this.

**Comment by pawan-idrive** [comment]
@diganta-dutta we we do have both of these in `vnc_auto.html`
```
        <div id="status">Loading</div>
        <div id="sendCtrlAltDelButton">Send CtrlAltDel</div>
```

File: vendor/idrive_vnc_base.css [hunk 5]
```diff
@@ -0,0 +1,86 @@
...
+  cursor: pointer;
+  width: auto;
+  float: right;
+  border: none;
+  background: #fff;
+  color: #0071bc;
+  border-radius: 2px
+}
+
+#sendClipBoard {
```
Comment (diganta-dutta) [inline vendor/idrive_vnc_base.css:50]: @pawan-idrive Sorry asked at the wrong location. Is this present?

**Comment by pawan-idrive** [comment]
@diganta-dutta I have removed this CSS code for `sendClipBoard` as it is not in use and merged the changes. Also moved the tag to latest master

**Pull request merged**